DesktopUI2/DesktopUI2/ConnectorHelpers.cs

```diff
@@ -61,12 +59,12 @@ public static async Task<Base> ReceiveCommit(Commit commit, StreamState state, P
           }
 
           //Treat all operation errors as fatal
-          throw new SpeckleException($"Failed to receive commit: {commit.id} objects from server", ex);
+          throw new SpeckleException($"Failed to receive commit: {commit.id} objects from server: {s}", ex);
         },
         onTotalChildrenCountKnown: c => progress.Max = c,
-        disposeTransports: true
+        disposeTransports: false
       )
-      .ConfigureAwait(true);
```

Comment:
Hey @JR-Morgan , any specific reason for these changes?

Reply:
The `ConfigureAwait(true)` was added during the reformat for all usages in DUI2 because we wanted to be extra safe about introducing breaking threading changes. In core we are using `ConfigureAwait(false)` everywhere since we don't care about what thread we execute on.

In this case, we are not doing any UI work that would necessitate us being on the UI thread, so (`ConfigureAwait(false)`) is desirable to avoid the costly synchronization of `ConfigureAwait(true)`

Reply:
And the case of `disposeTransports`, 
There are some issues with our `Operations.Recieve` and `Operations.Send` that mean that transports won't always be disposed (even when passing `disposeTransports: true`

imo, I don't really like how we even expose this as an option, it's far better to let the caller handle disposal, since they are the ones creating and storing the transport.

I've noted in #2392 a few improvements I think we could make to both Operations and transports, since these nitty-gritty details are not self-evident, nor are they a great DX.